### DISCUSSION TITLE
How can I tell if a TIF is a multi-page TIF file or a multi-graph TIF file

### DISCUSSION THREAD

**Discussion by Sc-Dz** [dlemstra/Magick.NET#1243] [Help] [answered]
How can I tell if a TIF is a multi-page TIF file or a multi-graph TIF file.

**Comment by dlemstra** [comment, marked as answer]
Can you explain what a multi-graph tif file is? And what you are trying to do?

**Reply by Sc-Dz** [reply]
I have found a solution


        public static bool HasMultiPageTIF(string filename)
        {
            if (!File.Exists(filename)) return false;

            try
            {
                TiffReadDefines tiffRead = new TiffReadDefines();

                tiffRead.IgnoreTags = new string[] { "ignore-layers" };

                MagickReadSettings readSettings = new MagickReadSettings(tiffRead);

                using (MagickImageCollection magimages = new MagickImageCollection())
                {
                    magimages.Ping(filename, readSettings);

                    if (magimages.Count > 1)
                    {
                        return true;
                    }
                    else
                    {
                        return false;
                    }
                }
            }
            catch (Exception)
            {
                return false;
            }
        }
